### PR TITLE
Implement `tsh proxy kube`

### PR DESCRIPTION
**Pull request by greedy52**
Implements #21870
RFD #22347

This is the basic `tsh proxy kube` implementation. Will do a separate change for TLS routing behind L7 LB improvement.

Usage 1:
```
$ tsh login ...
$ tsh kube login <cluster1>
$ tsh kube login <cluster2>
$ tsh proxy kube -p 8888
```

Usage 2:
```
$ tsh proxy kube <cluster1>
```

### CODE REVIEW

File: lib/kube/kubeconfig/localproxy.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	"github.com/gravitational/teleport/lib/srv/alpnproxy/common"
+)
+
+// LocalProxyCluster contains values for a kube cluster for generating
+// local proxy kubeconfig.
+type LocalProxyCluster struct {
+	// TeleportCluster is the Teleport cluster name.
+	TeleportCluster string
+	// KubeCluster is the Kubernetes cluster name.
+	KubeCluster string
```
[the remark this answers is not in the document]
Comment (greedy52) [reply lib/kube/kubeconfig/localproxy.go:34]: Good catch. Will remove `KubeClusters`. It's left over when copying `Values`.

File: lib/client/interfaces.go [hunk 2]
```diff
@@ -403,16 +403,34 @@ func (k *Key) TeleportTLSCertificate() (*x509.Certificate, error) {
...
+	}
+	keyPem, err := k.PrivateKey.RSAPrivateKeyPEM()
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	cert, err := keys.X509KeyPair(tlsCert, keyPem)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	return cert, nil
```
Comment (AntonAM) [inline lib/client/interfaces.go:431]: Maybe realign naming a bit for more clarity? (We also can apply `certPEM` inside `KubeTLSCert()`)
```suggestion
	certPEM, ok := k.KubeTLSCerts[kubeClusterName]
	if !ok {
		return tls.Certificate{}, trace.NotFound("TLS certificate for kubernetes cluster %q not found", kubeClusterName)
	}
	keyPem, err := k.PrivateKey.RSAPrivateKeyPEM()
	if err != nil {
		return tls.Certificate{}, trace.Wrap(err)
	}
	tlsCert, err := keys.X509KeyPair(certPEM, keyPem)
	if err != nil {
		return tls.Certificate{}, trace.Wrap(err)
	}
	return tlsCert, nil
```

**Review by AntonAM — COMMENTED** [review]
Great work! Just a few smaller comments.

File: tool/tsh/kube_proxy.go [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+	fp, err := alpnproxy.NewKubeForwardProxy(cf.Context, port, lp.GetAddr())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	kubeConfigPath := os.Getenv(proxyKubeConfigEnvVar)
+	if kubeConfigPath == "" {
+		_, port, _ := net.SplitHostPort(fp.GetAddr())
+		kubeConfigPath = path.Join(profile.KubeConfigPath(fmt.Sprintf("localproxy-%v", port)))
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply tool/tsh/kube_proxy.go:221]: I think not having to export again is a nice thing to have 👍

**Review by smallinsky — COMMENTED** [review]
First pass - I have left some suggestion but overall LGTM. I will do a second pass tomorrow.

**Review by AntonAM — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**